python/lsst/sims/movingObjects/linearObs.py

```diff
@@ -0,0 +1,293 @@
+from lsst.obs.lsstSim import LsstSimMapper
+from lsst.sims.coordUtils import chipNameFromRaDec
+
+from .orbits import Orbits
+from .ephemerides import PyOrbEphemerides
+
+__all__ = ['LinearObs', 'runLinearObs']
+
+
+class LinearObs(Orbits):
+    """
+    Class to generate observations of a set of moving objects.
+    Linear interpolation between gridpoint of ephemerides.
+    Inherits from Orbits to read orbits.
+    """
+    def __init__(self, orbitFile, delim=None, skiprows=None, ephfile=None,
+                 timescale='TAI', obscode=807):
+        super(LinearObs, self).__init__()
+        self.readOrbits(orbitfile=orbitFile, delim=None, skiprows=None)
+        self.pyephem = PyOrbEphemerides(ephfile=ephfile)
+        self.timescale = self.pyephem.timeScales[timescale]
+        self.obscode = obscode
+
+    def setTimesRange(self, timeStep=1., timeStart=49353., timeEnd=453003.):
+        """
+        Set an array for oorb of the ephemeris times desired, given the range of values.
+        @ timeStep : timestep for ephemeris generation (days)
+        @ timeStart : starting time of ephemerides (MJD)
+        @ timeEnd : ending time of ephemerides (MJD)
+        """
+        # Extend times beyond first/last observation, so that interpolation doesn't fail
+        timeStep = float(timeStep)
+        timeStart = timeStart - timeStep
+        timeEnd = timeEnd + timeStep
+        times = np.arange(timeStart, timeEnd + timeStep/2.0, timeStep)
+        # For pyoorb, we need to tag times with timescales;
+        # 1= MJD_UTC, 2=UT1, 3=TT, 4=TAI
+        self.ephTimes = np.array(zip(times, repeat(self.timescale, len(times))), dtype='double', order='F')
+
+    def setTimes(self, times):
+        """
+        Set an array for oorb of the ephemeris times desired, given an explicit set of times.
+        @ times : numpy array of the actual times of each ephemeris position.
+        """
+        self.ephTimes = np.array(zip(times, repeat(self.timescale, len(times))), dtype='double', order='F')
+
+
+    def generateEphs(self, sso):
+        """Generate ephemerides for all times in self.ephTimes.
+        """
+        self.pyephem.setOrbits(sso)
+        oorbEphs = self.pyephem._generateOorbEphs(self.ephTimes, obscode=self.obscode)
+        ephs = self.pyephem._convertOorbEphs(oorbEphs, byObject=True)
+        return ephs
+
+    # Linear interpolation
+    def interpolateEphs(self, ephs, i=0):
+        """
+        Generate linear interpolations between the quantities in ephs over time.
+        """
+        interpfuncs = {}
+        for n in ephs.dtype.names:
+            if n == 'time':
+                continue
+            interpfuncs[n] = interpolate.interp1d(ephs['time'][i], ephs[n][i], kind='linear',
+                                                  assume_sorted=True, copy=False)
+        return interpfuncs
+
+    def _setupCamera(self):
+        """
+        Initialize camera mapper, etc.
+        """
+        self.mapper = LsstSimMapper()
```

Comment:
I had kind of thought there would be one LinearObs instance in one invocation of python -- I think this class will be used as it is used in "make_linear_obs.py" rather than over and over.  I can swap it to lsst_camera though .. how do you recommend I make sure that all LinearObs instances are using the same lsst_camera object?